contracts/modules/validators/K1Validator.sol

```diff
@@ -236,12 +251,23 @@ contract K1Validator is IValidator, ERC7739Validator {
             return false;
         }
 
-        if (SignatureCheckerLib.isValidSignatureNowCalldata(owner, hash, signature)) {
-            return true;
+        // verify signer
+        try this.recoverSigner(hash, signature) returns (address recoveredSigner) {
+            if (recoveredSigner == owner) {
+                return true;
+            }
+        } catch {
+            // If recovery fails, we'll continue to the next check
         }
-        if (SignatureCheckerLib.isValidSignatureNowCalldata(owner, MessageHashUtils.toEthSignedMessageHash(hash), signature)) {
-            return true;
+
+        try this.recoverSignerFromEthSignedMessage(hash, signature) returns (address recoveredSigner) {
+            if (recoveredSigner == owner) {
+                return true;
+            }
+        } catch {
```

Comment:
since solady's `recover()` reverts if the returned address is zero, you need to make an external call to catch the revert. Instead, you can call [`tryRecover()`](https://github.com/Vectorized/solady/blob/main/src/utils/ECDSA.sol#L218) which returns zero address instead of reverting.

Using this function, this code becomes (just make sure `owner` isn't zero address):
```solidity
if(recoverSigner(hash, signature) == owner) return true;
if(recoverSignerFromEthSignedMessage(hash, signature) == owner) return true;
```

Reply:
I see. can do

Reply:
I also suggest that instead of `recoverSignerFromEthSignedMessage(hash, signature)` we can use recoverSigner(hash.toEthSignedMessageHash(), signature)